collect_app/src/main/java/org/odk/collect/android/utilities/DateTimeUtils.java

```diff
@@ -104,19 +104,19 @@ private static String getCustomDateTimeLabel(Date date, DatePickerDetails datePi
         return String.format(context.getString(R.string.custom_date), customDateText, gregorianDateText);
     }
 
-    private static String getDateTimePattern(boolean containsTime, DatePickerDetails datePickerDetails) {
-        String datePattern;
+    private static String getDateTimeSkeleton(boolean containsTime, DatePickerDetails datePickerDetails) {
+        String dateSkeleton;
         if (containsTime) {
-            datePattern = "yyyyMMMdd HHmm";
+            dateSkeleton = "yyyyMMMMdd HHmm";
```

Comment:
Why do we need this change in months? Isn't it because before you used full month names? (the previous commit) now it would be unnecessary I think.

Reply:
The Method `getBestDateTimePattern` on line 45 to which the skeleton is passed return the Best Possible Date Time Pattern depending upon the locale. During testing, I found out that the function returns shorter month names even with `MMMM` in the `dateSkeleton`
Refer -> https://developer.android.com/reference/android/text/format/DateFormat#getBestDateTimePattern(java.util.Locale,%20java.lang.String)
The only reason I added an `MMMM` instead of  `MMM` is that `MMM` are not as common in other locales especially the vernacular ones as much as in EN(US) and MMMM is a still a standard. The output remains the same for EN(US) as I have tested but it may give better results for other locales

Reply:
Quoting the Docs

> public static String getBestDateTimePattern (Locale locale,
> String skeleton)
> 
> Returns the best possible localized form of the given skeleton for the given locale. A skeleton is similar to, and uses the same format characters as, a Unicode UTS#35 pattern.
> 
> One difference is that order is irrelevant. For example, "MMMMd" will return "MMMM d" in the en_US locale, but "d. MMMM" in the de_CH locale.
> 
> Note also in that second example that the necessary punctuation for German was added. For the same input in es_ES, we'd have even more extra text: "d 'de' MMMM".
> 
> This method will automatically correct for grammatical necessity. Given the same "MMMMd" input, this method will return "d LLLL" in the fa_IR locale, where stand-alone months are necessary. Lengths are preserved where meaningful, so "Md" would give a different result to "MMMd", say, except in a locale such as ja_JP where there is only one length of month.
> 
> This method will only return patterns that are in CLDR, and is useful whenever you know what elements you want in your format string but don't want to make your code specific to any one locale.

Reply:
Ok, I see.